app/src/main/java/org/mozilla/fenix/home/intent/OrientationLockIntentProcessor.kt

```diff
@@ -0,0 +1,26 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.home.intent
+
+import android.content.Intent
+import android.content.pm.ActivityInfo
+import androidx.appcompat.app.AppCompatActivity
+import androidx.navigation.NavController
+import mozilla.components.browser.state.engine.middleware.EXTRA_REQUESTED_ORIENTATION
+
+class OrientationLockIntentProcessor(private val activity: AppCompatActivity) : HomeIntentProcessor {
```

Comment:
Drive by - Can you add a kDoc for what this class does

Reply:
Sure, documentation and tests to be added.

Just wanted first to validate the general flow which both here and in AC seems a bit too complex but it gets the job done while respecting the reactive arch and prevents uglier implementations with maybe a weak reference to activity and allows for communication of simple events not necessarily part of a MVI state.